include/seqan/blast/blast_record.h

```diff
@@ -0,0 +1,384 @@
+
+        alignStats.numGaps              = std::numeric_limits<unsigned>::max();
+        alignStats.numGapOpens          = std::numeric_limits<unsigned>::max();
+        alignStats.numGapExtensions     = std::numeric_limits<unsigned>::max();
+        alignStats.numInsertions        = std::numeric_limits<unsigned>::max();
+        alignStats.numDeletions         = std::numeric_limits<unsigned>::max();
+        alignStats.numMatches           = std::numeric_limits<unsigned>::max();
+        alignStats.numMismatches        = std::numeric_limits<unsigned>::max();
+        alignStats.numPositiveScores    = std::numeric_limits<unsigned>::max();
+        alignStats.numNegativeScores    = std::numeric_limits<unsigned>::max();
+        alignStats.alignmentLength      = std::numeric_limits<unsigned>::max();
+        alignStats.alignmentSimilarity  = std::numeric_limits<float>::max();
+        alignStats.alignmentIdentity    = std::numeric_limits<float>::max();
+        alignStats.alignmentScore       = std::numeric_limits<unsigned>::max();
+
+        clear(align.data_rows);
+    }
+};
+
+inline bool
+_memberIsSet(CharString const & in)
+{
+    return (in != "not init");
+}
+
+template <typename TNumber>
+inline SEQAN_FUNC_ENABLE_IF(Is<NumberConcept<TNumber> >, bool)
+_memberIsSet(TNumber const & in)
+{
+    return (in != std::numeric_limits<TNumber>::max());
+}
+
+template <typename TQId,
+          typename TSId,
+          typename TPos,
+          typename TAlign>
+inline void
+clear(BlastMatch<TQId, TSId, TPos, TAlign> & match)
+{
+    match._clear();
+}
+
+/*!
+ * @class BlastRecord
+ * @implements FormattedFileRecordConcept
+ * @headerfile <seqan/blast.h>
+ * @signature struct BlastRecord<TQId, TSId, TPos, TAlign> { ... };
+ * @brief A record of blast-matches (belonging to one query).
+
+ * @tparam TQId  Type of qId, defaults to std::string
+ * @tparam TSId  Type of sId, defaults to std::string
+ * @tparam TPos  Position type of the sequences, defaults to <tt>uint32_t</tt>
+ * @tparam TAlign Type of the @link Align @endlink member, defaults to
+ * <tt>Align<CharString, ArrayGaps></tt>
+ */
+
+template <typename TQId = std::string,
+          typename TSId = std::string,
+          typename TPos = uint32_t,
+          typename TAlign = Align<std::string, ArrayGaps>>
+struct BlastRecord
+{
+    /*!
+     * @typedef BlastRecord::TBlastMatch
+     * @signature typedef BlastMatch<TQId, TSId, TPos, TAlign> TBlastMatch;
+     * @brief type of the contained matches
+     */
+    typedef         BlastMatch<TQId, TSId, TPos, TAlign> TBlastMatch;
+
+    /*!
+     * @var TQId BlastRecord::qId;
+     * @brief verbose Id of the query
+     */
+    TQId            qId;
+
+    /*!
+     * @var TPos BlastRecord::qLength;
+     * @brief length of the query sequence
+     */
+    TPos            qLength;
+
+    /*!
+     * @var std::list<TBlastMatch> BlastRecord::matches;
+     * @brief list of the contained matches
+     */
+    std::list<TBlastMatch>  matches;
```

Comment:
> Do you insert elements in the middle?

No, but swapping is instant in a linked list, becaus only the pointers to the elements are changed. So for fat obejcts this is better than swapping in a vector.